lib/optimizely/optimizely_user_context.rb

```diff
@@ -69,6 +70,11 @@ def set_attribute(attribute_key, attribute_value)
     # @return [OptimizelyDecision] A decision result
 
     def decide(key, options = nil)
+      unless @fetch_segments_thread.nil?
```

Comment:
Pure sync/async both will provide an option that they can skip decide on error, but not with this thread/join. Wondering if there is a simple solution to support :) We can chat.
`
segments = fetch_segments();
if segments != null {
   decide();
}
`
`
fetch_segments(non-blocking);
decide();
`